src/rsa.rs

```diff
@@ -333,14 +360,21 @@ impl RSAKeyPair {
     /// platforms, it is done less perfectly. To help mitigate the current
     /// imperfections, and for defense-in-depth, base blinding is always done.
     /// Exponent blinding is not done, but it may be done in the future.
-    pub fn sign(&self, padding_alg: &'static RSAPadding,
+    pub fn sign(&self, padding_alg: &'static pad::PaddingAlgorithm,
                 rng: &rand::SecureRandom, msg: &[u8], signature: &mut [u8])
                 -> Result<(), ()> {
         if signature.len() != self.public_modulus_len() {
             return Err(());
         }
 
-        try!(padding_alg.pad(msg, signature));
+        try!(padding_alg.encode(msg, signature, rng));
+
+        let mod_bits = self.public_modulus_len_bits() - 1;
```

Comment:
I am happy to restrict RSA signing to public moduli that are exact multiples of 8 bits.

Regardless of that, do whatever needs to be done to move this logic to `encode()`, whether that means adding a new parameter or whatever.